src/requests/adapters.py

```diff
@@ -494,6 +500,7 @@ def send(
                 retries=self.max_retries,
                 timeout=timeout,
                 chunked=chunked,
+                **self.urllib3_response_options,
```

Comment:
I don't believe a `TypeError` here is actually a good user experience, especially if we want this to be something we get into a relatively soon release (because this has never raised a `TypeError` and if it starts to do so now, it is not something users will be prepared to handle - hence why I suggest overriding arguments with what's specified here already.